### DISCUSSION TITLE
Raspberry Pi hangs running CLI tool

### DISCUSSION THREAD

**Discussion by jpmiller25** [immich-app/immich#823] [Q&A]
Hello, can yall help me diagnose why this is happening? I'm running this on a headless raspberry pi, and connecting through ssh to run the cli tool. I have the immich cli tool version 0.22.0.

I'm running the command
`immich upload --yes --email ###@gmail.com --password "###" --server http://192.168.1.25:2283/api -d /mnt/hdd/sftp/photo_import/`

It recognizes 2017 assets, and says it will upload 2015 assets. Then it gets started but only gets to 14% before it hangs, and then after a while my ssh session disconnects. I'm working remotely so I can't manually reboot, so I just wait a few hours and usually I can reconnect and try again.

Is there a log stored somewhere that I can check what's actually going on? Thanks!

**Comment by alextran1502** [comment]
I assume you also running Immich on the Pi, and you are trying to upload the existing files from your Pi to the Immich instance? 

If that is true, your Pi is probably get choked with all the processing tasks. 

You can set the CLI to use one thread with the argument `--thread 1` to reduce the amount of asset upload in parallel and thus reducing the CPU usage

**Reply by jpmiller25** [reply]
OK, yes that's probably what it is.  I'll try it.

**Reply by jpmiller25** [reply]
OK, yes that's probably what it is.  I'll try it.

**Reply by jpmiller25** [reply]
I tried it with 1 thread and it reached 16% before hanging.

**Reply by alextran1502** [reply]
What is your Pi spec?

**Reply by jpmiller25** [reply]
Pi 4b, 4gb ram. 
Immich volume and folder to be uploaded are both on the same usb3.1 HDD. 
I'm running docker on Ubuntu server 22.04 64bit.

**Reply by jpmiller25** [reply]
So I just keep trying to run it and each time it gets a bit further before it hangs and disconnects the ssh session. I'm letting it continue to run though, after a few hours typically I can log in and try again without having to hard reboot.

It's strange though that every time it starts it says there are 2017 assets, and 2015 to upload. It's like it's not picking up the photos that have already been uploaded.

**Reply by alextran1502** [reply]
Can you try disable the machine learning container by commented out that section in the docker-compose file? It might help with the CPU usage

**Reply by jpmiller25** [reply]
OK, I'll try that and update, thanks

**Comment by jpmiller25** [comment]
I'm watching top as it runs, and node definitely runs the most CPU %, but mount.ntfs is occupying almost equal cpu percent as node.
Is it a mistake to be storing the data on an ntfs drive?

**Comment by jpmiller25** [comment]
either way, looks like that made a big difference, its now at 45% and I haven't seen it get anywhere near that far up to this point. Thanks!

**Comment by jpmiller25** [comment]
So I have a terminal up running the immich cli tool and another running top. Immich froze this time at 54%, but the top terminal is still connected and running. The node process is now only running 0.4% cpu. top processes now are kworker/0:0+kblockd (17%cpu), and kswap0 (15%cpu). I'm noticing memory available is down to 80MiB, but swap says 0 total, 0 free, 0 used.

**Reply by jpmiller25** [reply]
So I'm still not sure what's going on here. I decided to move my installation to an old laptop I have running an amd64 processor. It's still only 4gb ram, but I'll see if the new installation solves this.

**Reply by alextran1502** [reply]
Do you have updates on this?

**Reply by jpmiller25** [reply]
I'm still working on it, need to migrate the database to a new server tonight or tomorrow then I'll test it.

**Reply by jpmiller25** [reply]
After moving my installation to an old laptop running ubuntu server 22.04, I was able to run the CLI tool and get it to finish without hanging. 
The cli tool finds 2192 assets, 1864 to be uploaded to the server. It takes a while to go through, but returns with 5 failed uploads, with status code 504. details here https://pastebin.com/cdWQ5hj9
After it finishes, running it again starts with 2192 assets, 1864 to upload again so if there is supposed to be a function to avoid reupload it's not working.
